### PR TITLE
Update MIGraphx Docker to point to latest working onnxrt branch.

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Will use this as a staging branch between ORT builds and updating CI on our end to catch any integration issues between our EP and MIGraphX

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
Related to onnxruntime issue mentioned here: https://github.com/microsoft/onnxruntime/issues/14126

File: Dockerfile [hunk 1]
```diff
@@ -98,3 +99,23 @@ ENV UBSAN_OPTIONS=print_stacktrace=1
...
+    hipcub  \
+    hipblas  \
+    hipify-clang \
+    half && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# add this for roctracer dependancies
+RUN pip3 install CppHeaderParser
+RUN pip3 install packaging==22.0
```
Comment (pfultz2) [inline Dockerfile:121]: Combine this into one line: `pip3 install CppHeaderParser packaging==22.0`.

File: Dockerfile [hunk 2]
```diff
@@ -98,3 +99,23 @@ ENV UBSAN_OPTIONS=print_stacktrace=1
...
+    rocthrust \
+    rocrand \
+    hipsparse \
+    rccl \
+    rccl-dev \
+    rocm-smi-lib \
+    roctracer-dev \
+    hipcub  \
+    hipblas  \
+    hipify-clang \
```
Comment (pfultz2) [inline Dockerfile:114]: Why do we need hipify?

File: Dockerfile (hunk 1 above)
Comment (TedThemistokleous) [reply Dockerfile:121]: Okay will do

File: Dockerfile [hunk 3]
```diff
@@ -98,3 +99,23 @@ ENV UBSAN_OPTIONS=print_stacktrace=1
 ENV ASAN_OPTIONS=detect_stack_use_after_return=1:check_initialization_order=1:strict_init_order=1
 RUN ln -s /opt/rocm/llvm/bin/llvm-symbolizer /usr/bin/llvm-symbolizer
 
+#Setup HIPIFY and rocRAND libraries (used for Onnxruntime)
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Dockerfile:103]: Will move this up

File: Dockerfile [hunk 4]
```diff
@@ -72,14 +76,11 @@ RUN /download_models.sh && rm /download_models.sh
 # Install latest ccache version
 RUN cget -p $PREFIX install facebook/zstd@v1.4.5 -X subdir -DCMAKE_DIR=build/cmake
 RUN cget -p $PREFIX install ccache@v4.1 -DENABLE_TESTING=OFF
-
-# Install newer cmake for onnx runtime
-ARG CMAKE_VERSION=3.24.2
-RUN cget -p /opt/cmake install -X binary https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
+RUN cget -p $PREFIX install kitware/cmake@v3.24.3 -DCMAKE_USE_OPENSSL=OFF
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Dockerfile:79]: Oh I was just doing similar to the 5.4 docker changes.

File: Dockerfile (hunk 2 above)
Comment (TedThemistokleous) [reply Dockerfile:114]: To build onnxrt. last build failed since it was looking for a hip compiler for the MIGraphX EP

**Comment by TedThemistokleous** [comment]
Seem to still be hitting this issue in the CI now,

```
[2023-01-13T00:58:43.836Z] -- Build files have been written to: /onnxruntime/build/Linux/Release/_deps/abseil_cpp-subbuild

[2023-01-13T00:58:44.254Z] [ 11%] Creating directories for 'abseil_cpp-populate'

[2023-01-13T00:58:44.254Z] [ 22%] Performing download step (download, verify and extract) for 'abseil_cpp-populate'

[2023-01-13T00:58:44.254Z] -- Downloading...

[2023-01-13T00:58:44.254Z]    dst='/onnxruntime/build/Linux/Release/abseil-cpp/src/20211102.0.zip'

[2023-01-13T00:58:44.254Z]    timeout='none'

[2023-01-13T00:58:44.254Z]    inactivity timeout='none'

[2023-01-13T00:58:44.254Z] -- Using src='https://github.com/abseil/abseil-cpp/archive/refs/tags/20211102.0.zip'

[2023-01-13T00:58:44.254Z] CMake Error at /onnxruntime/build/Linux/Release/abseil-cpp/src/abseil_cpp-populate-stamp/download-abseil_cpp-populate.cmake:170 (message):

[2023-01-13T00:58:44.254Z]   Each download failed!

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z]     error: downloading 'https://github.com/abseil/abseil-cpp/archive/refs/tags/20211102.0.zip' failed

[2023-01-13T00:58:44.254Z]           status_code: 1

[2023-01-13T00:58:44.254Z]           status_string: "Unsupported protocol"

[2023-01-13T00:58:44.254Z]           log:

[2023-01-13T00:58:44.254Z]           --- LOG BEGIN ---

[2023-01-13T00:58:44.254Z]           Protocol "https" not supported or disabled in libcurl

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z]   Closing connection -1

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z]   

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z]           --- LOG END ---

[2023-01-13T00:58:44.254Z]           

[2023-01-13T00:58:44.254Z]     

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z] 

[2023-01-13T00:58:44.254Z] make[2]: *** [CMakeFiles/abseil_cpp-populate.dir/build.make:100: /onnxruntime/build/Linux/Release/abseil-cpp/src/abseil_cpp-populate-stamp/abseil_cpp-populate-download] Error 1

[2023-01-13T00:58:44.254Z] make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/abseil_cpp-populate.dir/all] Error 2
```

Locally same issue even when I try to build a 5.4 rocm version. Guessing there's some weird order of operations thats breaking libcurl?

**Comment by causten** [comment]
I opened https://github.com/microsoft/onnxruntime/issues/14293

**Comment by TedThemistokleous** [comment]
> I opened [microsoft/onnxruntime#14293](https://github.com/microsoft/onnxruntime/issues/14293)

Looks like it was the openssl issue that you mentioned. I was able to get builds running now. Just going to see what CI says now, and then move up the onnxrt branch until we're tracking main.

**Review by causten — APPROVED** [review]

**Comment by TedThemistokleous** [comment]
Enabling tests for this locally, looks like we're failing on the two tests that the Onnxrt team excludes from their CI in

tools/ci_build/gitbhub/pai/migraphx-excluded-tests.txt

Do we want to add the same/is there a decent way to add these exceptions on our end?

The two tests that are failing are :

GatherOpTest.Gather_invalid_index_cpu
Scatter.InvalidIndex

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (causten) [reply Dockerfile]: I believe it means you would replace $PREFIX with /opt/cmake.

[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Dockerfile]: Okay, I'll change this.

[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Dockerfile]: Done

File: Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply Dockerfile]: Done

File: Dockerfile (hunk 1 above)
Comment (TedThemistokleous) [reply Dockerfile]: Done

**Comment by TedThemistokleous** [comment]
> Enabling tests for this locally, looks like we're failing on the two tests that the Onnxrt team excludes from their CI in
> 
> tools/ci_build/gitbhub/pai/migraphx-excluded-tests.txt
> 
> Do we want to add the same/is there a decent way to add these exceptions on our end?
> 
> The two tests that are failing are :
> 
> GatherOpTest.Gather_invalid_index_cpu Scatter.InvalidIndex

@causten @pfultz2  got this running their CI now and grabbing their excluded tests directly from the onnxrt repo made in the container. 

Latest commit splits out the target commit sha-1 to a file so its easier for a github action maintain and modify the .onnxrt-commit file. Using the  "." infront here to avoid triggering license checker issues.

File: Dockerfile [hunk 5]
```diff
@@ -72,19 +93,21 @@ RUN /download_models.sh && rm /download_models.sh
...
+RUN cget -p /opt/cmake install kitware/cmake@v3.24.3
 
-# Install newer cmake for onnx runtime
-ARG CMAKE_VERSION=3.24.2
-RUN cget -p /opt/cmake install -X binary https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
 
 ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
 ARG ONNXRUNTIME_BRANCH=main
-ARG ONNXRUNTIME_COMMIT=24f1bd6156cf5968bbc76dfb0e801a9b9c56b9fc
+RUN export ONNXRUNTIME_COMMIT=$(cat onnxrt-commit.txt)
```
Comment (causten) [inline Dockerfile:101]: You sure about the file name?  .onnxrt-commit vs onnxrt-commit.txt

File: tools/build_and_test_onnxrt.sh [hunk 6]
```diff
@@ -26,5 +26,8 @@ pip3 install -r requirements.txt
 # Add newer cmake to the path
 export PATH="/opt/cmake/bin:$PATH"
 export CXXFLAGS="-D__HIP_PLATFORM_HCC__=1 -w"
-./build.sh --config Release --update --build --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --test --use_migraphx
-# pip3 install /code/onnxruntime/build/Linux/Release/dist/*.whl
+./build.sh --config Release  --cmake_extra_defines CMAKE_HIP_COMPILER=/opt/rocm/llvm/bin/clang++ --update --build --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --skip_tests --rocm_home /opt/rocm --use_migraphx --migraphx_home /opt/rocm --rocm_version=`cat /opt/rocm/.info/version-dev`
```
Comment (causten) [inline tools/build_and_test_onnxrt.sh:29]: This skips test.  The script use to --test.  If the fix for the failing tests is coming in a different PR, add a TODO: Turn tests back on with #xxxx

**Review by causten — CHANGES_REQUESTED** [review]

File: tools/build_and_test_onnxrt.sh (hunk 6 above)
Comment (TedThemistokleous) [reply tools/build_and_test_onnxrt.sh:29]: It still builds onnxruntime_test_all

Invoking the migraphx_test_launcher.sh is what's done in the build.py script if --test is set. I confirmed this locally and verified tests are run before I pushed this.

File: Dockerfile (hunk 5 above)
Comment (TedThemistokleous) [reply Dockerfile:101]: Fixing this. Thanks!

**Comment by TedThemistokleous** [comment]
Should be good to go. Opened #1530 for the cron based scheduled task weekly that will open a PR

File: Dockerfile [hunk 7]
```diff
@@ -72,19 +93,23 @@ RUN /download_models.sh && rm /download_models.sh
...
+RUN cget -p /opt/cmake install kitware/cmake@v3.24.3
 
-# Install newer cmake for onnx runtime
-ARG CMAKE_VERSION=3.24.2
-RUN cget -p /opt/cmake install -X binary https://github.com/Kitware/CMake/releases/download/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION}-Linux-x86_64.tar.gz
 
 ARG ONNXRUNTIME_REPO=https://github.com/Microsoft/onnxruntime
 ARG ONNXRUNTIME_BRANCH=main
-ARG ONNXRUNTIME_COMMIT=24f1bd6156cf5968bbc76dfb0e801a9b9c56b9fc
+RUN export ONNXRUNTIME_COMMIT=$(cat test/onnx/.onnxrt-commit)
```
Comment (pfultz2) [inline Dockerfile:101]: The `ONNXRUNTIME_COMMIT` should be an `ARG` so we can build the dockerfile with a different onnx version without needing to modify the dockerfile.

Comment (TedThemistokleous) [reply Dockerfile]: Changed back to ARG

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**